api/internal/localizer/localizer.go

```diff
@@ -33,6 +35,10 @@ type localizer struct {
 
 	// destination directory in newDir that mirrors root
 	dst string
+
+	// copiedHelmDefaultHome indicates whether the default helm chart home has
+	// been copied to dst at this root
+	copiedHelmDefaultHome bool
```

Comment:
Instead of this being a field on `localizer`, can we have a function that checks if a particular directory already exists and use that instead? 

The problem I'm foreseeing is someone using the `HelmChartInflationGenerator` directly through the `generators` field (I know that's not part of this PR, but the next one), and they specify the same chartHome in multiple of those generators. In that case, we will need a similar to check to make sure that we don't attempt to copy the same directory multiple times.

Reply:
In the next PR, I'm planning to simply call the `copyChartHome` function. That should handle checking this field without any additional code, right?

Reply:
Done. Thanks for the explanation.

We now use existence as an indicator for having previously copied the chart home. We accept that in an edge case this may lead to partially copied directories: https://github.com/kubernetes-sigs/kustomize/pull/4978/files#diff-c21f7b3d0afda4a64163411e90e72d2fed6c3df14c7b6f8ed07237bb5e1fc91cR1286-R1305
We also choose this method over a map field in `localizer` that stores which chart homes we've copied.